### ISSUE TITLE
Memory leak in SSL sockets

### ISSUE THREAD

**Issue by vyzo** [gambit/gambit#344]
Using an SSL socket leaks a still refcounted (foreign) object per socket.

The following program demonstrates:
```
(declare (block) (standard-bindings) (extended-bindings))

(define request
  (string-append "GET / HTTP/1.1\r\n"
                 "User-Agent: Mozilla/5.0 (compatible)\r\n"
                 "Accept: */*\r\n"
                 "Connection: close\r\n"
                 "\r\n"))

(define (https-get host)
  (let ((sock
         (open-tcp-client (list server-address: host
                                port-number: 443
                                eol-encoding: 'cr-lf
                                tls-context: (make-tls-context)))))
    (write-substring request 0 (string-length request) sock)
    (force-output sock)
    (thread-sleep! 1)
    (close-port sock)))

(define (displayln what)
  (display what)
  (newline))

;; still refcounted object counting
(c-declare #<<END-C
// these are defined in gambit/lib/mem.c
#define ___PSTATE_MEM(var) ___ps->mem.var
#define still_objs ___PSTATE_MEM(still_objs_)
#define ___STILL_LINK_OFS 0
#define ___STILL_REFCOUNT_OFS 1
#define ___STILL_BODY_OFS 6
#define ___STILL_HAND_OFS ___STILL_BODY_OFS
END-C
)

(define count-still-objects/refcount
  (c-lambda () int #<<END-C
int count = 0;
___WORD *base = ___CAST(___WORD*,still_objs);
while (base != 0)
{
 if (base[___STILL_REFCOUNT_OFS])
  {
   count++;
  }
 base = ___CAST(___WORD*,base[___STILL_LINK_OFS]);
}
___return(count);
END-C
))


(define (main host)
  (displayln (count-still-objects/refcount))
  (https-get host)
  (displayln (count-still-objects/refcount))
  (thread-sleep! 1)
  (https-get host)
  (displayln (count-still-objects/refcount)))

(main (cadr (command-line)))

```

To compile and run:
```
$ gsc -exe -o ssl-leak ssl-leak.scm
$ ./ssl-leak www.google.com
1
2
3
```

cc @fare

**Comment by vyzo** [comment]
After reading through the code, and confirming with some quick experimentation in the repl, it seems it's the tls context that's leaking:
```
> (count-still-objects/refcount)
1
> (make-tls-context)
#<foreign #4 0x23aa760>
> (count-still-objects/refcount)
2
```

**Comment by vyzo** [comment]
Simple fix in #345.
